src/Areas/Server/Commands/ToolLoading/CommandFactoryToolLoader.cs

```diff
@@ -184,35 +185,23 @@ private static Tool GetTool(string fullName, IBaseCommand command)
 
         var options = command.GetCommand().Options;
 
-        var schema = new JsonObject
-        {
-            ["type"] = "object"
-        };
+        var schema = new ToolInputSchema();
 
         if (options != null && options.Count > 0)
         {
-            var arguments = new JsonObject();
             foreach (var option in options)
             {
-                arguments.Add(option.Name, new JsonObject()
+                schema.Properties.Add(option.Name, new ToolPropertySchema
                 {
-                    ["type"] = option.ValueType.ToJsonType(),
-                    ["description"] = option.Description,
+                    Type = option.ValueType.ToJsonType(),
+                    Description = option.Description,
                 });
             }
 
-            schema["properties"] = arguments;
-            schema["required"] = new JsonArray(options.Where(p => p.IsRequired).Select(p => (JsonNode)p.Name).ToArray());
+            schema.Required = options.Where(p => p.IsRequired).Select(p => p.Name).ToArray();
         }
-        else
-        {
-            var arguments = new JsonObject();
-            schema["properties"] = arguments;
-        }
-
-        var newOptions = new JsonSerializerOptions(McpJsonUtilities.DefaultOptions);
 
-        tool.InputSchema = JsonSerializer.SerializeToElement(schema, new JsonSourceGenerationContext(newOptions).JsonNode);
```

Comment:
here we avoid pulling in `JsonNode` converters by using a statically defined type.